packages/headless-react/typedoc/assets/vars/dark-mode-toggle.js

```diff
@@ -0,0 +1,360 @@
+    this._permanentCheckbox = shadowRoot.querySelector(
+      '[part=permanentCheckbox]'
+    );
+    this._permanentLabel = shadowRoot.querySelector('[part=permanentLabel]');
+
+    // Does the browser support native `prefers-color-scheme`?
+    const hasNativePrefersColorScheme = matchMedia(MQ_DARK).media !== NOT_ALL;
+    // Listen to `prefers-color-scheme` changes.
+    if (hasNativePrefersColorScheme) {
+      matchMedia(MQ_DARK).addListener(({matches}) => {
+        this.mode = matches ? DARK : LIGHT;
+        this._dispatchEvent(COLOR_SCHEME_CHANGE, {colorScheme: this.mode});
+      });
+    }
+    // Set initial state, giving preference to a remembered value, then the
+    // native value (if supported), and eventually defaulting to a light
+    // experience.
+
+    const rememberedValue = this.areFunctionalCookiesEnabled
+      ? store.getItem(NAME)
+      : '';
+    if (rememberedValue && [DARK, LIGHT].includes(rememberedValue)) {
+      this.mode = rememberedValue;
+      this._permanentCheckbox.checked = true;
+      this.permanent = true;
+    } else if (hasNativePrefersColorScheme) {
+      this.mode = matchMedia(MQ_LIGHT).matches ? LIGHT : DARK;
+    }
+    if (!this.mode) {
+      this.mode = LIGHT;
+    }
+    if (
+      this.permanent &&
+      !rememberedValue &&
+      this.areFunctionalCookiesEnabled
+    ) {
+      store.setItem(NAME, this.mode);
+    }
+
+    // Default to toggle appearance.
+    if (!this.appearance) {
+      this.appearance = TOGGLE;
+    }
+
+    // Update the appearance to either of toggle or switch.
+    this._updateAppearance();
+
+    // Update the radios
+    this._updateRadios();
+
+    // Make the checkbox reflect the state of the radios
+    this._updateCheckbox();
+
+    // Synchronize the behavior of the radio and the checkbox.
+    [this._lightRadio, this._darkRadio].forEach((input) => {
+      input.addEventListener('change', () => {
+        this.mode = this._lightRadio.checked ? LIGHT : DARK;
+        this._updateCheckbox();
+        this._dispatchEvent(COLOR_SCHEME_CHANGE, {colorScheme: this.mode});
+      });
+    });
+    this._darkCheckbox.addEventListener('change', () => {
+      this.mode = this._darkCheckbox.checked ? DARK : LIGHT;
+      this._updateRadios();
+      this._dispatchEvent(COLOR_SCHEME_CHANGE, {colorScheme: this.mode});
+    });
+
+    // Make remembering the last mode optional
+    this._permanentCheckbox.addEventListener('change', () => {
+      this.permanent = this._permanentCheckbox.checked;
+      this._dispatchEvent(PERMANENT_COLOR_SCHEME, {
+        permanent: this.permanent,
+      });
+    });
+
+    // Finally update the mode and let the world know what's going on
+    this._updateMode();
+    this._dispatchEvent(COLOR_SCHEME_CHANGE, {colorScheme: this.mode});
+    this._dispatchEvent(PERMANENT_COLOR_SCHEME, {
+      permanent: this.permanent,
+    });
+  }
+
+  attributeChangedCallback(name, oldValue, newValue) {
```

Comment:
You don't seem to use `oldValue`
```suggestion
  attributeChangedCallback(name, newValue) {
```

Reply:
Good catch! Thanks JP!

Reply:
⚠️ the suggested code is invalid. 
```suggestion
  attributeChangedCallback(name, _oldValue, newValue) {
```

You can't just "omit" a parameter; order matters.